api/e2e/new-test/e2e_test.go

```diff
@@ -0,0 +1,418 @@
+package e2e
```

Comment:
Why is the filename different than the internal one? Can we keep the same names so it's easier to compare later? (Similar comment about other files.)